discovery/gossiper.go

```diff
@@ -2441,6 +2414,54 @@ func (d *AuthenticatedGossiper) processNetworkAnnouncement(
 	}
 }
 
+// processZombieUpdate determines whether the provided channel update should
+// resurrect a given zombie edge.
+func (d *AuthenticatedGossiper) processZombieUpdate(
+	chanInfo *channeldb.ChannelEdgeInfo, msg *lnwire.ChannelUpdate) error {
+
+	// The least-significant bit in the flag on the channel update tells us
+	// which edge is being updated.
+	isNode1 := msg.ChannelFlags&lnwire.ChanUpdateDirection == 0
+
+	// Since we've deemed the update as not stale above, before marking it
+	// live, we'll make sure it has been signed by the correct party. If we
+	// have both pubkeys, either party can resurect the channel. If we've
+	// already marked this with the stricter, single-sided resurrection we
+	// will only have the pubkey of the node with the oldest timestamp.
+	var pubKey *btcec.PublicKey
+	switch {
+	case isNode1 && chanInfo.NodeKey1Bytes != emptyPubkey:
+		pubKey, _ = chanInfo.NodeKey1()
+	case !isNode1 && chanInfo.NodeKey2Bytes != emptyPubkey:
+		pubKey, _ = chanInfo.NodeKey2()
+	}
+	if pubKey == nil {
+		return fmt.Errorf("incorrect pubkey to resurrect zombie "+
```

Comment:
Looks like we can just return nil here? Otherwise this will get logged as an error when it really isn't.

Reply:
If we return nli then the case statement will fall through to normal processing, but we really want to cease processing this update if the wrong pubkey is used since we don't want to allow that to resurrect things.

Reply:
Ah I see, let's change that log to the warn level then? This case will likely be hit very often as the other side is constantly updating their channels.

Reply:
Good call on log-level, could also make this is a concrete error that we filter out.